worker.go

```diff
@@ -93,12 +99,14 @@ func startWorkers(fileName string, nbWorkers int, env PreparedEnv) error {
 
 				// TODO: make the max restart configurable
 				if _, ok := workersRequestChans.Load(absFileName); ok {
-					workersReadyWG.Add(1)
 					if fc.exitStatus == 0 {
+						workersReadyWG.Add(1)
```

Comment:
This is still WIP, there's an issue where the waitgroup will never resolve if the worker script itself has an error. Moving this line here makes it more consistent, but doesn't fully resolve it yet.

Reply:
Ah, nice find. That's probably because there is only one `workersReadyWG.Done()` called (when this function ends) but we `.Add(1)` every time the worker restarts.

Reply:
I was able to resolve this by properly ignoring the waitgroup after the initial load